src/main/java/org/nixos/idea/lsp/ui/CommandSuggestionsPopup.java

```diff
@@ -110,8 +109,8 @@ public void caretUpdate(CaretEvent e) {
     }
 
     private final class MyListPopup extends ListPopupImpl implements JBPopupListener {
-        private MyListPopup() {
-            super(null, new MyListPopupStep());
+        private MyListPopup(List<Suggestion> suggestions) {
+            super(null, new MyListPopupStep(suggestions));
```

Comment:
polish: You shouldn't need to add the suggestions as constructor arguments here. These classes are _inner classes_. They can access the fields from the outer class.